### PR TITLE
gtk 3.24.24

### PR DESCRIPTION
**Pull request by ericLemanissier**
this isa port of https://github.com/bincrafters/conan-gtk

Specify library name and version:  **gtk/3.24.24**

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/gtk/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    def requirements(self):
+        self.requires("gdk-pixbuf/2.42.0")
+        self.requires("glib/2.67.0")
+        if self.settings.compiler != "Visual Studio":
+            self.requires("cairo/1.17.2")
+        if self.settings.os == "Linux":
+            self.requires("at-spi2-atk/2.38.0")
+            if self.options.with_wayland:
+                self.requires("xkbcommon/0.10.0")
+                self.requires("wayland") # FIXME: Create an actual Wayland package(s)
```
Comment (Croydon) [inline recipes/gtk/all/conanfile.py:71]: TODO: Figuring out which parts of Wayland gtk needs and what packages we need accordingly

**Comment by Croydon** [comment]
The stale bot starts to annoy me, maybe we should extend the times to 2 months for a warning and 3 months for closing?

File: recipes/gtk/all/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/gtk/all/conanfile.py:70]: the last missing one is [wayland-protocols](https://github.com/GNOME/gtk/blob/3.24.22/meson.build#L552) which is blocked by https://github.com/conan-io/conan-center-index/pull/3663
wayland-client, wayland-cursor and wayland-egl are already packaged by https://conan.io/center/wayland

**Comment by SSE4** [comment]
> The stale bot starts to annoy me, maybe we should extend the times to 2 months for a warning and 3 months for closing?

feel free to create PR to update stale bot settings as you wish. the purpose is not to annoy anyone, but instead to help us, so let's tune it to be helpful, not annoying.

**Comment by ericLemanissier** [comment]
@jgsogo @danimtb is C3I drowning under the snow ?

**Comment by Croydon** [comment]
Madrid has actually quite a situation right now. Stay safe everyone ❤️

**Comment by ericLemanissier** [comment]
[Beware the dinosaurs !](https://twitter.com/arrola/status/1347884489452773376)

**Comment by ericLemanissier** [comment]
call for reviews!

@madebr @SpaceIm @prince-chrismc @Croydon @intelligide @theirix @gocarlos @uilianries @SSE4

**Review by SSE4 — APPROVED** [review]

File: recipes/gtk/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Linux":
+            if self.options.with_wayland or self.options.with_x11:
+                if not self.options.with_pango:
+                    raise ConanInvalidConfiguration("with_pango option is mandatory when with_wayland or with_x11 is used")
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("GTK recipe is not yet compatible with Windows. Contributions are welcome.")
```
Comment (madebr) [inline recipes/gtk/all/conanfile.py:59]: What are the problems with Windows?

**Review by madebr — APPROVED** [review]

File: recipes/gtk/all/conanfile.py (hunk 2 above)
Comment (ericLemanissier) [reply recipes/gtk/all/conanfile.py:59]: the first blocking one is that ninja fails unless you use env var `CONAN_USE_ALWAYS_SHORT_PATHS=1` (`short_paths` attribute is not enough)

File: recipes/gtk/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir.replace("gtk", "gtk+"), self._source_subfolder)
+
+    def _configure_meson(self):
+        meson = Meson(self)
+        defs = {}
+        if self.settings.os == "Linux":
+            defs["wayland_backend"] = "true" if self.options.with_wayland else "false"
+            defs["x11_backend"] = "true" if self.options.with_x11 else "false"
+        defs["introspection"] = "false"
```
Comment (prince-chrismc) [inline recipes/gtk/all/conanfile.py:93]: Could this be `False`?

File: recipes/gtk/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        with tools.environment_append({"PKG_CONFIG_PATH": self.install_folder}):
+            meson.install()
+
+        self.copy(pattern="COPYING", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.components["gdk-3.0"].libs = ["gdk-3"]
+        self.cpp_info.components["gdk-3.0"].includedirs = [os.path.join("include", "gtk-3.0")]
+        self.cpp_info.components["gdk-3.0"].requires = []
```
Comment (prince-chrismc) [inline recipes/gtk/all/conanfile.py:125]: Is this required?

**Review by danimtb — APPROVED** [review]

**Comment by Croydon** [comment]
Fantastic work @ericLemanissier! This really required some heavy lifting 💪

File: recipes/gtk/all/conanfile.py (hunk 3 above)
Comment (madebr) [reply recipes/gtk/all/conanfile.py:93]: No, meson is very lucky about casings.

**Review by madebr — APPROVED** [review]

**Comment by ericLemanissier** [comment]
merge commit has a green tick, but no binary are available on https://conan.io/center/gtk?version=3.24.24&os=Linux&tab=configuration nor when running `conan search gtk/3.24.24@`
What's happening with CI ?

**Comment by prince-chrismc** [comment]
Missing packages again... same as #4251

**Comment by jgsogo** [comment]
Having a look to it. There are some packages available when running `conan search gtk/3.24.24@ -r conan-center --table=gtk.html`. I'll dig into the logs to check what happened to the others

**Comment by jgsogo** [comment]
Failure on our side (library implementation). Something like (sure you spot the bug):

```
result = UNKNOWN
try {
    conan_build
    result = OK
    conan_upload
}
catch() {
    result = ERROR
}
finally() {
   set_build_status(result)
}
```

I did blame in the repository and the author of that commit is already pushing the wheel of pain at our offices

image

**Comment by prince-chrismc** [comment]
Feels like an interview questions!

```diff
try {
    conan_build
-   result = OK
    conan_upload
+   result = OK
}
```

I'm available 😉

**Comment by jgsogo** [comment]
> I'm available 😉

@prince-chrismc , We can talk about it if you want to. :grimacing:

**Pull request merged**